### DISCUSSION TITLE
Remote client?

### DISCUSSION THREAD

**Discussion by NickRahn** [GregHib/void#550] [Q&A] [answered]
If i am hosting the server on a server and want to play from another client how would i update the client file to point to a remote server. hopefully via IP Address

**Comment by GregHib** [comment, marked as answer]
Yes you can change the IP Address from 127.0.0.1 to your desired remote server located inside the [client's loader class](https://github.com/GregHib/void-client/blob/72fabcdc715b063a4fc7a22d1b66d811898a7ed9/client/src/Loader.java#L53). You can then rebuild it into a jar and use it as you wish

**Reply by Somadevv** [reply]
Hey I'm struggling to get the correct 'Main class' for rebuilding the new Jar with the changes made, could you let me know how to do this correctly? Thanks in advance!

When I create a new Jar and run it, i get the following:

/Library/Java/JavaVirtualMachines/jdk-25.jdk/Contents/Home/bin/java -Dfile.encoding=UTF-8 -Dsun.stdout.encoding=UTF-8 -Dsun.stderr.encoding=UTF-8 -jar /Users/joshi]Documents/Osrs/out/artifacts/Osrs_jar/Osrs.jar
Error: Could not find or load main class 
Caused by: java.lang.ClassNotFoundException: 

Process finished with exit code 1

**Reply by GregHib** [reply]
> Hey I'm struggling to get the correct 'Main class' for rebuilding the new Jar with the changes made, could you let me know how to do this correctly? Thanks in advance!
> 
> When I create a new Jar and run it, i get the following:
> 
> /Library/Java/JavaVirtualMachines/jdk-25.jdk/Contents/Home/bin/java -Dfile.encoding=UTF-8 -Dsun.stdout.encoding=UTF-8 -Dsun.stderr.encoding=UTF-8 -jar /Users/joshi]Documents/Osrs/out/artifacts/Osrs_jar/Osrs.jar Error: Could not find or load main class Caused by: java.lang.ClassNotFoundException:
> 
> Process finished with exit code 1

The Main Class is `Loader` however I find that this error is often more due to the META-INF file location, which will hopefully be resolved by overwriting it's location to a higher level in the project.

<img width="685" height="732" alt="image" src="https://github.com/user-attachments/assets/6c00ed56-42f3-4233-b919-39acc1832dd2" />
